modules/apps/headless-apio/forms/forms-apio-client-test/src/testIntegration/java/com/liferay/forms/apio/client/test/TextFormApioTest.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.forms.apio.client.test;
+
+import static com.liferay.forms.apio.client.test.matcher.FormApioTestMatchers.isBoolean;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+import com.liferay.forms.apio.client.test.internal.activator.TextFormApioTestBundleActivator;
+import com.liferay.forms.apio.client.test.util.FormApioTestUtil;
+import com.liferay.oauth2.provider.test.util.OAuth2ProviderTestUtil;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import java.util.Map;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.Archive;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Cruz
+ */
+@RunAsClient
+@RunWith(Arquillian.class)
+public class TextFormApioTest {
+
+	@Deployment
+	public static Archive<?> getArchive() throws Exception {
+		return OAuth2ProviderTestUtil.getArchive(
+			TextFormApioTestBundleActivator.class);
+	}
+
+	@Before
+	public void setUp() throws MalformedURLException {
+		_rootEndpointURL = new URL(_url, "/o/api");
+	}
+
+	@Test
+	public void testGetTextFieldFromForm() {
+		Map<String, Object> fieldProperties =
+			FormApioTestUtil.getFieldProperties(
```

Comment:
When you read this test, you only read: FormApioTestUtil.getFieldProperties(
 				_rootEndpointURL, _TEXT_FIELD_NAME)

You need to much so much from how FormApioTestUtil.getFieldProperties works to understand what it is really going on in this test :(.

It is better If you keep the full request in the test, and the asserts also.

Reply:
I partially agree, I understood that with this signature it would be not fully understandable, but IMHO we should reuse the requests to get the field properties as they will be the same for all field tests. It will a be bit hard to maintain the code for all the fields and scenarios (translations, data providers, rules, ...) when the API changes.

Reply:
I undestand totally your point, but I can't agree :(.

I also don't see the point in test every scenario using client test, IMHO we should move all the complex scenarios to integration tests.

When functional test fails it is really hard to know why, so we should move all we can as unit or integration tests.

Reply:
I'll try to rewrite the tests to explicit the steps and update here.